brave-zipkin-spancollector/src/main/java/com/github/kristofa/brave/zipkin/ZipkinSpanCollector.java

```diff
@@ -73,20 +72,24 @@ public ZipkinSpanCollector(final String zipkinCollectorHost, final int zipkinCol
         final ZipkinSpanCollectorParams params) {
         Validate.notEmpty(zipkinCollectorHost);
         Validate.notNull(params);
-        clientProvider =
-            new ZipkinCollectorClientProvider(zipkinCollectorHost, zipkinCollectorPort, params.getSocketTimeout());
-        try {
-            clientProvider.setup();
-        } catch (final TException e) {
-            if (params.failOnSetup()) {
-                throw new IllegalStateException(e);
-            } else {
-                LOGGER.warn("Connection could not be established during setup.", e);
-            }
-        }
+
         spanQueue = new ArrayBlockingQueue<Span>(params.getQueueSize());
         executorService = Executors.newFixedThreadPool(params.getNrOfThreads());
+
         for (int i = 1; i <= params.getNrOfThreads(); i++) {
+
+            //Creating a client provider for every spanProcessingThread.
```

Comment:
It would be good if you could extract the client creation in a separate private method. This would make the constructor easier to read.